### DISCUSSION TITLE
Map voting

### DISCUSSION THREAD

**Discussion by u1ps** [etlegacy/etlegacy#1939] [Ideas]
Right now the map voting works like this if i am correct?

You get list of maps to choose from and you can pick up to 3 maps and give 3 votes with different points (1, 2 and 3). If you consider 3 - 100% , 2 - 66%, 3 - 33%. Lets say there are 3 maps on the list you would like to play. Your option is to choose only one and give it 100% and 2nd vote will be 66%, 3th 33% impact. Or you give this 1 pt to map "i might want to play?".

Would it not be better to give 1 equal point up to 3 maps from list. This way you can choose up to 3 maps you want to play and each vote has equal impact. It feels much easier for player to vote maps like that. Might be just my opinion.

My idea would be having check boxes in front of each map on the  voting list. Each player can have up to 3 boxes checked and after sending votes each checked box will count as 1pt.

I think up to 3 votes, each vote equal 1pts might work better.
![312123rey6t](https://user-images.githubusercontent.com/99591943/162626145-56b6eb89-fd59-4adf-afb7-c2f10e0dd97d.png)

**Comment by u1ps** [comment]
Could someone specify the source code locationr for map voting menu, i can try and test with it.

**Reply by Aranud** [reply]
Here you go https://github.com/etlegacy/etlegacy/blob/a3e9b56eb4d504920671efb34af4b247f52a6ecc/src/cgame/cg_debriefing.c#L1122

**Comment by Aciz** [comment]
Agree with the proposed changes here - I'd even be happy with giving only 1 vote per client. Current system favors popular maps too much due to them likely getting multiple 2/1 point votes anyway, just because they are popular.

**Reply by u1ps** [reply]
There are lots of line in code to rework it. Most simple way to achieve the proposed change would be to make every single vote equal to 1pt. (I think this is essential). Now about multiple votes i dont mind it as long as every vote is equal to 1pt. Perhaps test and try it in next patch without reworking it too much. 

1. option:  1 vote per client equal to 1 pt
2. option: up to 3 votes per client, each equal to 1pt

I prefer option if there are 2 maps in list id like to play i can vote fro them equally. In the end if every player would participate in voting the next map should be preference of majority.

Suggestion: try fiddle with 2 variables in next patch? easy to change and revert back if needed.

- each vote = 1pt.
- amount of votes per client= x (1-3)
